independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/utils/BuildToolHelper.java

```diff
@@ -0,0 +1,80 @@
+package io.quarkus.bootstrap.utils;
+
+import static io.quarkus.bootstrap.util.QuarkusModelHelper.DEVMODE_REQUIRED_TASKS;
+
+import io.quarkus.bootstrap.resolver.AppModelResolverException;
+import io.quarkus.bootstrap.resolver.QuarkusGradleModelFactory;
+import io.quarkus.bootstrap.resolver.model.QuarkusModel;
+import io.quarkus.bootstrap.util.QuarkusModelHelper;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * Helper class used to expose build tool used by the project
+ */
+public class BuildToolHelper {
+
+    public enum BuildTool {
+        MAVEN("pom.xml"),
+        GRADLE("build.gradle");
+
+        private final String buildFile;
+
+        BuildTool(String buildFile) {
+            this.buildFile = buildFile;
+        }
+
+        public String getBuildFile() {
+            return buildFile;
+        }
+    }
+
+    private BuildToolHelper() {
+
+    }
+
+    public static boolean isMavenProject(Path project) {
+        Path currentPath = project;
+        while (currentPath != null) {
+            if (Files.exists(currentPath.resolve(BuildTool.MAVEN.getBuildFile()))) {
+                return true;
+            }
+            currentPath = currentPath.getParent();
```

Comment:
Did you copy this from somewhere? This is not a good idea. What kind of path is expected to be passed in? Project root or any path inside a project?
A little bit better version would be checking each directory for all the recognizable build files instead of walking up to the root of the FS for each build file.

Reply:
yes I copier that from here: https://github.com/quarkusio/quarkus/blob/42dbdd05cd20ef1e82ad62e6617a48e8ea9100c8/independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/workspace/LocalProject.java#L217

Reply:
Ok, that path is derived from a class name (that is in the classes dir of a project) or is a project root.